pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstorecache_test.go

```diff
@@ -0,0 +1,444 @@
+		size:     3,
+		items:    list.New(),
+		mutex:    sync.RWMutex{},
+		itemsMap: make(map[string]MapItem),
+	}
+	got := NewCache(3)
+	require.Equal(t, want, got, "NewCache() = %v, want %v", got, want)
+}
+
+func TestCacheimpl_GetSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	m[selectors1.Key] = MapItem{
+		item:    &selectors1,
+		element: items.PushFront(selectors1.Key),
+	}
+	m[selectors2.Key] = MapItem{
+		item:    &selectors2,
+		element: items.PushFront(selectors2.Key),
+	}
+
+	cacheInstance := &cacheImpl{
+		size:     3,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name         string
+		want         *Item
+		key          string
+		errorMessage string
+	}{
+		{
+			name:         "Non existing entry",
+			want:         nil,
+			key:          selectors3.Key,
+			errorMessage: "A non-existing item's key should return a nil item.",
+		},
+		{
+			name:         "Existing entry",
+			want:         &selectors1,
+			key:          selectors1.Key,
+			errorMessage: "An existing items key's should return the existing item",
+		},
+	}
+
+	for _, tt := range tests {
+		got := cacheInstance.GetSignature(tt.key)
+		require.Equal(t, got, tt.want, "%v Got: %v Want: %v", tt.errorMessage, got, tt.want)
+	}
+}
+
+func TestCacheimpl_PutSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	cacheInstance := &cacheImpl{
+		size:     2,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name       string
+		item       *Item
+		wantLength int
+		wantKey    string
+		wantValue  *Item
+	}{
+		{
+			name:       "Put first element",
+			item:       &selectors1,
+			wantLength: 1,
+			wantKey:    selectors1.Key,
+			wantValue:  &selectors1,
+		},
+		{
+			name:       "Put first element again",
+			item:       &selectors1,
+			wantLength: 1,
+			wantKey:    selectors1.Key,
+			wantValue:  &selectors1,
+		},
+		{
+			name:       "Put second element",
+			item:       &selectors2,
+			wantLength: 2,
+			wantKey:    selectors2.Key,
+			wantValue:  &selectors2,
+		},
+		{
+			name:       "Overflow cache",
+			item:       &selectors3,
+			wantLength: 2,
+			wantKey:    selectors3.Key,
+			wantValue:  &selectors3,
+		},
+		{
+			name:       "Update entry",
+			item:       &selectors3Updated,
+			wantLength: 2,
+			wantKey:    selectors3.Key,
+			wantValue:  &selectors3Updated,
+		},
+	}
+
+	for _, tt := range tests {
+		cacheInstance.PutSignature(*tt.item)
+		gotLen := cacheInstance.items.Len()
+		if gotLen != tt.wantLength {
+			t.Errorf("Item count should be %v in test case %q", tt.wantLength, tt.name)
+		}
+		gotItem, present := m[tt.wantKey]
+		if !present {
+			t.Errorf("Key put but not found: %v", tt.wantKey)
+		}
+		require.Equal(t, gotItem.item, tt.wantValue, "Value different than expected. \nGot: %v \nWant:%v", gotItem.item, tt.wantValue)
+	}
+}
+
+func TestCacheimpl_CheckOverflowAndUpdates(t *testing.T) {
```

Comment:
this approch can result on unexpected issues, and at the same time contains steps that are not requied.
In the end you want to test overflow and in a full list how it is working, why not to create a full list and put a new one dending in what you want to test?